packages/plugin-core/src/test/suite-integ/NativeTreeView.test.ts

```diff
@@ -0,0 +1,314 @@
+import { expect } from "../testUtilsv2";
+import { describe } from "mocha";
+import { EngineNoteProvider } from "../../views/EngineNoteProvider";
+import { describeMultiWS } from "../testUtilsV3";
+import { MockEngineEvents } from "./MockEngineEvents";
+import { NoteProps } from "@dendronhq/common-all";
+import { ExtensionProvider } from "../../ExtensionProvider";
+import { RenameNoteV2aCommand } from "../../commands/RenameNoteV2a";
+import { vault2Path } from "@dendronhq/common-server";
+import { Uri } from "vscode";
+import path from "path";
+import { VSCodeUtils } from "../../vsCodeUtils";
+import { NoteTestUtilsV4 } from "@dendronhq/common-test-utils";
+
+function getNoteUri(opts: { note: NoteProps; wsRoot: string }) {
+  const { note, wsRoot } = opts;
+  const { fname, vault } = note;
+  const notePath = fname + ".md";
+  const vaultPath = vault2Path({ vault, wsRoot });
+  return Uri.file(path.join(vaultPath, notePath));
+}
+
+async function runRenameNote(opts: { noteId: string; newName: string }) {
+  const engine = ExtensionProvider.getEngine();
+  const { wsRoot } = engine;
+
+  const { noteId, newName } = opts;
+  const noteToRename = engine.notes[noteId];
+  const noteToRenameVaultPath = vault2Path({
+    wsRoot,
+    vault: noteToRename.vault,
+  });
+  const rootUri = Uri.file(noteToRenameVaultPath);
+  const oldUri = getNoteUri({ note: noteToRename, wsRoot });
+  const newUri = VSCodeUtils.joinPath(rootUri, `${newName}.md`);
+  const renameCmd = new RenameNoteV2aCommand();
```

Comment:
Out of curiosity - when are we used to use `RenameNoteV2aCommand` vs `MoveNoteCommand`?  We should rename the command soon, "V2a" 😅

Reply:
I actually don't have a good answer to that haha. both do their own thing but the end result is the same 🤷🏼